test/interfaces/IAllowanceTransfer.sol

```diff
@@ -0,0 +1,165 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.17;
+
+import {IEIP712} from "./IEIP712.sol";
+
+/// @title AllowanceTransfer
+/// @notice Handles ERC20 token permissions through signature based allowance setting and ERC20 token transfers by checking allowed amounts
+/// @dev Requires user's token approval on the Permit2 contract
+interface IAllowanceTransfer is IEIP712 {
```

Comment:
is this not importable from permit2 dep?

Reply:
Unfortunately Permit2 has solidity pinned to `=0.8.17` preventing forge from building it together with the rest of calibur (`^0.8.23`). I had to go with a similar approach to UniswapX where we copy in these interfaces and have a custom Permit2 deploy function.

Reply:
but the interfaces are not pinned

Reply:
as long as you don't import Permit2.sol I don't think it should be an issue, I just tried it here https://github.com/Uniswap/calibur/pull/229

Reply:
Awesome, thanks @snreynolds